### ISSUE TITLE
[New Feature RC.apps] Api to set admin configs.

### ISSUE THREAD

**Issue by ear-dev** [WideChat/Rocket.Chat#953]
Currently the only versioning we have is in settings.ts, but each of our deployments uses unique configs for some values, and we would like to be able to configure them programmatically with an api.

**Comment by ear-dev** [comment]
- Use current rocketchat API to get App IDs (already exist?)
- Create a new RC API to set config values for specific App IDs ( need to implement )

**Comment by ear-dev** [comment]
@AlexanderKanakis Please show examples of how we would use your new api to set the settings......  please show the different flavors.....  json file for config, string values, any other variety. 

Include the workflow, i.e. first we need to make a call to get the appID, then use the appID in the subsequent calls?

**Comment by AlexanderKanakis** [comment]
PR: https://github.com/WideChat/Rocket.Chat/pull/981

## Proposed changes 
Added functionality for:

1. Getting app ID with the name of the APP.
2. Changing app setting with setting ID and app ID.
3. Changing app setting with setting ID and app name.

### Functions:

1. `getAppIdByName(appName: string)`: Retuns App ID
2. `updateAppSettingById(appId: string, settingId: string, value: any)`: Changes the value of a setting using app ID. 

- settingID can be found in `App/config/Settings.ts`  ![image](https://user-images.githubusercontent.com/18405180/138768781-0ab4375a-c0ad-454f-95e9-112bff4af1cc.png)
- value can be number, string, or json object/array.

3. `updateAppSettingByName(appName: string, settingId: string, value: any)`: Changes the value of a setting using app name. 

## Examples:

 1. Inside an async function
```
const appId = await getAppIdByName('Dialogflow');

await updateAppSettingById(appId, 'dialogflow_customer_timeout_time', 240);

```
2. Inside a sync function
```
const newDFVersions = [
            { key: 'ES', i18nLabel: 'ES' },
            { key: 'CX', i18nLabel: 'CX' },
        ]
Promise.await( updateAppSettingByName( 'Dialogflow', 'dialog_flow_version', newDFVersions ) )
```

**Comment by ear-dev** [comment]
@Shailesh351 can you review and test please?

**Comment by ear-dev** [comment]
@AlexanderKanakis in light of the new task to remove all global configs from the DF.app, and to instead use a config.json for each DF.agent endpoint which describes the config entirely, this API work will have to be adjusted for the new design once it is ready.

**Comment by ear-dev** [comment]
@AlexanderKanakis as we discussed just now.......  you will change dialogflow_bot_list_config to 'agents'.

**Comment by ear-dev** [comment]
@AlexanderKanakis All of the other rocketchat rest APIs are nested under "/api/v1/*/"  Is there a reason that these new APIs are not also under "v1'?  Can we standardize on that?

Also.......  do we actually need "appname" in the url?  For example:  http://localhost:3000/api/apps/appname/Dialogflow/settings/agents/mexico_agent  

Is it enough to just use apps/Dialogflow/*?  Or is there some reason you need to nest down through "appname"?

Oh....  this is to distinguish between "appname" and "appID" isn't it?

**Comment by AlexanderKanakis** [comment]
@ear-dev  The already existing AppsRestApi provided by Rocket.Chat nests the app REST API under `/api/apps/*/`

The call `apps/Dialogflow/*` would fall under the route `apps/ ${ appID }`, which accepts the app ID and not app name. We can remove that route if we will always use the app name for the REST call and use `apps/Dialogflow/*`

**Comment by ear-dev** [comment]
@AlexanderKanakis perhaps we should not use appByName at all, and just stick with appID.  What do you think?  It could be that that makes more sense, it's more explicit, and prevents any confusion....

I'm also thinking that we might want to move the whole thing under v1 for consistency.

I think after you have created an openAPI spec that describes this api, it will facilitate a design meeting where we can more easily discuss these types of details.

**Comment by ear-dev** [comment]
@AlexanderKanakis Please paste an example of an RC.app entry in the RC database here.  I would like to see the full schema representation of a Rocketchat app, as it exists in the mongodb.
